integration-test/Analysis/FixtureExpectationUtils.hs

```diff
@@ -91,17 +93,17 @@ withProjectOfType result (projType, projPath) =
   find (\(dr, _) -> projectType dr == projType && projectPath dr == projPath) result
 
 testSuiteHasSomeDepResults :: (AnalyzeProject a, Show a, Eq a) => AnalysisTestFixture a -> DiscoveredProjectType -> Spec
-testSuiteHasSomeDepResults fixture projType = aroundAll (withAnalysisOf fixture) $
+testSuiteHasSomeDepResults fixture projType = aroundAll (withAnalysisOf NonStrict fixture) $
   describe (toString $ testName fixture) $ do
     it "should find targets" $ \(result, extractedDir) ->
       expectProject (projType, extractedDir) result
 
     it "should have some dependency results" $ \(result, extractedDir) ->
       isJust (getDepResultsOf result (projType, extractedDir)) `shouldBe` True
 
-testSuiteDepResultSummary :: (AnalyzeProject a, Show a, Eq a) => AnalysisTestFixture a -> DiscoveredProjectType -> DependencyResultsSummary -> Spec
-testSuiteDepResultSummary fixture projType depResultSummary =
-  aroundAll (withAnalysisOf fixture) $
+testSuiteDepResultSummary :: (AnalyzeProject a, Show a, Eq a) => Mode -> AnalysisTestFixture a -> DiscoveredProjectType -> DependencyResultsSummary -> Spec
+testSuiteDepResultSummary mode fixture projType depResultSummary =
+  aroundAll (withAnalysisOf mode fixture) $
     describe (toString $ testName fixture) $ do
       it "should find targets" $ \(result, extractedDir) ->
         expectProject (projType, extractedDir) result
```

Comment:
@csasarak Added some tests like how you suggested. Is this approach somewhat similar to what you had in mind?

Reply:
I was thinking more of something that first ran the test with `NonStrict` and then one that ran it with `Strict` and compared the results. Ideally, they should be the same for these tests because we always are testing the ideal case regardless of the flag. The idea here is that if the `NonStrict` test is always doing ideal analysis, then the `StrictMode` version should return the same results. Basically what I'm trying to think about is "How do we know that setting `Strict` actually does anything?". 

However, on second thought I think maybe the answer is to make `guardStrictMode` just a little bit smarter and test it in isolation in our unit tests:

```haskell
guardStrictMode :: (Has Diagnostics sig m) => Mode -> [m a] -> m a
guardStrictMode _ [] = fatalText "Strict mode: No Possible actions to guard against."
guardStrictMode NonStrict (action1:rest) = foldr (<||>) action1 rest
guardStrictMode Strict (action1:_) = action1 <||> fatalText "Strict mode enabled, skipping other actions"
```
with usage:
```haskell
guardStrictMode mode [tactic1, tactic2, tactic3, ...]
```
Testing then can be something more trivial:

```haskell
do it' "test1" $ do
       res <- recover $ guardStrictMode Strict [fatalText 1, pure 2]
       res `shouldBe'` Nothing
     it' "test2" $ do
       res <- recover $ guardStrictMode NonStrict [fatalText 1, pure 2]
       res `shouldBe'` Just 2
```
I haven't actually run the above spec, but I think it'd be something similar to that. I think this strikes a good balance by having at least some testing around the `Mode` behavior while also not requiring re-running expensive integration tests multiple times.

Reply:
Hey, I was thinking a little bit more about this and I don't think there's anything more you need to do - I think that as long as you've manually tested this it's OK.